dask/array/core.py

```diff
@@ -661,6 +661,22 @@ def map_blocks(
 
     original_kwargs = kwargs
 
+    # issue warning when mapping function with axis argument when array chunks > 1
+    if has_keyword(func, "axis"):
+        if len(arrs) == 1:
+            n_axis_chunks = [len(x) for x in arrs[0].chunks]
+            if any([n_chunks > 1 for n_chunks in n_axis_chunks]):
+                if "axis" in original_kwargs.keys():
+                    axis_chunks = n_axis_chunks[original_kwargs["axis"]]
+                    if axis_chunks > 1:
+                        warnings.warn(
+                            "Operating along axis with nchunks>1 may produce unexpected behavior"
+                        )
```

Comment:
I think this can be simplified some. Perhaps something like:

```suggestion
    if has_keyword(func, "axis") and "axis" in original_kwargs:
        axis = original_kwargs["axis"]
        if len(arrs) == 1:
            if arrs[0].chunksize[axis] != arrs[0].shape[axis]:
                warnings.warn(
                    "Operating along axis with nchunks>1 may produce unexpected behavior"
                )
```